byzcoin/tx_pipeline.go

```diff
@@ -327,13 +330,26 @@ func (p *txPipeline) processTxs(txChan <-chan ClientTransaction, initialState *t
 		p.wg.Add(1)
 		defer p.wg.Done()
 		intervalChan := getInterval()
+		var txHashes [][]byte
+	leaderLoop:
 		for {
 			select {
 			case tx, ok := <-txChan:
 				if !ok {
 					log.Lvl3("stopping txs processor")
 					return
 				}
+				for _, txHash := range txHashes {
```

Comment:
Using a map would be more efficient probably.

Reply:
but then it's more difficult to remove the oldest elements. In this case, I can simply drop the front of the slice. With a map, I would have to go through the whole map and decide for each element if it is too old.